tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -372,7 +371,7 @@ def _validate_is_arm_resource(self):
 
     def _validate_event_matches_resource(self):
         resource_type = self.policy.resource_manager.resource_type.resource_type
-        if resource_type is not GenericArmResource.resource_type.resource_type:
+        if resource_type != 'armresource':
```

Comment:
this check seems like it could be simplified, ie
```python
if instance(self.policy.resource_manager, GenericArmResource):
```
or
```python
if self.policy.resource_type == 'armresource':
```

Reply:
could you clarify how your triggering a test failure or issue that needs this change?

Reply:
This didn't trigger any tests failures, however it was failing for `c7n run` for almost any policy.

The issue is that we have our own `Notify` action with slightly different syntax. However, by default AWS Notify action is added to all resources.. So, if we import `GenericArmResource` in `policy.py`, we get 2 resources available: `GenericArmResource` and `ResourceGroup`. They both have `Notify` action, but these actions have different schemas (because `GenericArmResource` is not following Azure init code due to us not loading it explicitly).

As a result, there is an exception of different schemas for same action.

I'd treat this PR as a hotfix, but we should make some better change.

Reply:
Is this a scenario a generic test could verify?  (don't block fixing master on it, but we can file issue).

Reply:
I don't think it is easy to verify in a unittest since it might depend on tests order..